### DISCUSSION TITLE
Executing input() after Kathara.get_instance().connect_tty() : EOFError

### DISCUSSION THREAD

**Discussion by JH-UCLouvain** [Kidel/Kathara#295] [Q&A] [answered]
Hello ! I'm a new user of Kathara and the python API. I'm trying to do a python3 script where I run a custom client to interact with the different devices from the lab. Basically it's just a loop where I do an `input()` so the user can specify the device where he wants to connect with `Kathara.get_instance().connect_tty()`. After we exit from a device, we return to the main client with the next iteration of the loop, then it executes the next `input()` etc. The problem is that I got a `EOFError` with the `input()` when I try to execute it after the `connect_tty()` (it seems that it makes a conflict with the stdin or something like that). Is there a solution or an alternative for this problem ? Thank you in advance ! Here is my code:

```python
def run_client(self):
    print("Starting lab ...")
    Kathara.get_instance().deploy_lab(lab=self.lab)
    print("Enter the router name to connect to it or \"exit\".")
    print("When you are connected to a device, you can enter \"exit\" to return to the main client.")
    while True:
        entered = input("> ")
        if entered == "exit":
            return
        m_exists = False
        for m in self.lab.machines.items():
            if entered == m[0]: m_exists = True
        if not m_exists:
            print(f"Error : no device named \"{entered}\"")
        else:
            print(f"Connecting to device \"{entered}\" ...")
            Kathara.get_instance().connect_tty(entered, lab_name=self.lab.name)
``` 

The error I got:

```
Starting lab ...
Enter the router name to connect to it or "exit".
When you are connected to a device, you can enter "exit" to return to the main client.
> as35r1
Connecting to device "as35r1" ...
root@as35r1:/# exit
exit
> Traceback (most recent call last):
  File "/Kathara-BGP.py", line 78, in <module>
    ex.run_client()
  File "../Common/utils.py", line 98, in run_client
    entered = input("> ")
EOFError
```

**Comment by JH-UCLouvain** [comment, marked as answer]
I got a solution by re-opening the stdin:

```python
sys.stdin = open("/dev/tty")
user_input = input("> ")
```

**Reply by Skazza94** [reply]
Hi @JH-UCLouvain,
the `connect_tty` method redirects the system I/O through the `pyuv` library.

Probably, when the connection is closed, the I/O is not restored to the initial value, hence creating the exception.

The solution that you proposed is correct.

Thanks,
Mariano.
